zerver/tests/test_queue_worker.py

```diff
@@ -423,13 +420,15 @@ def fake_publish(queue_name: str, event: Dict[str, Any], processor: Optional[Cal
                        return_value=fake_response), \
                     mock_queue_publish('zerver.lib.queue.queue_json_publish',
                                        side_effect=fake_publish), \
-                    patch('logging.info'), \
+                    self.assertLogs(level='INFO') as m, \
                     self.settings(MAILCHIMP_API_KEY='one-two',
                                   PRODUCTION=True,
                                   ZULIP_FRIENDS_LIST_ID='id'):
                 worker.start()
 
         self.assertEqual(data['failed_tries'], 1 + MAX_REQUEST_RETRIES)
+        log_output = 'INFO:root:Processing signup for user 10 in realm zulip'
+        self.assertEqual(m.output, [log_output for i in range(4)])
```

Comment:
Perhaps `[log_output]*4` is a better syntax here?

Reply:
@chdinesh1089 I think `assertIn` won't be a very good idea here because it will try to match the whole string which varies with each test run and from system to system.
I think using `assertRegexpMatches` we can easily assert the required part, I have tried it for some logs I they were running fine.
**For Example:** Instead of:
`        self.assertEqual(logs[1][:36] + logs[1][-23:], 'INFO:root:Importing realm data from /test-export/realm.json')`
I used regular expression to get same result like this:
`self.assertRegexpMatches(logs[1], r'^INFO:root:Importing realm data from .*/test-export/realm.json$')`
If you agree with using it, I would go for making for similar change at other places also.

Reply:
@chdinesh1089 should I go for using `assertRegexpMatches` instead of `assertIn` and `string slicing for such logs`?

Reply:
I think that's okay.